src/groupeddataframe/groupeddataframe.jl

```diff
@@ -323,6 +323,21 @@ function Base.to_index(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return Base.to_index(gd, Tuple(key))
 end
 
+function _dict_to_tuple(key::Union{Dict{Symbol},Dict{<:AbstractString}} , gd)
+    if length(key) != length(gd.cols)
+        throw(KeyError(key))
+    end
+
+    T = keytype(key)
```

Comment:
This is needed as for strings we need to have a type that is able to perform the conversion described below
```suggestion
    T = keytype(key) <: Symbol ? Symbol : String
```
This assumes that a `Dict` with any string keys should be able to handle getting `String` input (@nalimilan - do you think it is safe to assume; maybe some strange custom `AbstractStrings` with non UTF-8 encoding would fail here?).